demos/common/python/models/detection_model.py

```diff
@@ -0,0 +1,118 @@
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+from .image_model import ImageModel
+from .utils import load_labels, clip_detections
+
+
+class DetectionModel(ImageModel):
+    '''A abstract detection model class
+
+    This class supports the detection models. The Detection Model should have single image input.
+
+    Attributes:
+        labels(List[str]): list of labels for classes (could be None)
+        threshold(float): threshold for detection filtering, any detection with confidence less than this value
+            should be omitted in ``posptrocess`` method (0<=thresold<=1.0 for most models)
+        iou_threshold(float): threshold for NMS detection filtering
+    '''
+
+    def __init__(self, ie, model_path, resize_type='standart', keep_aspect_ratio=False,
+                 labels=None, threshold=None, iou_threshold=None):
+        '''The Detection Model constructor
+
+        Calls the ``ImageModel`` construtor first.
+
+        Args:
+            labels(Iterable[str], str, Path): list of labels for detection classes or path to file with them
+            threshold(float): threshold for detections filtering by confidence
+            iou_threshold(float): threshold for NMS filtering
+
+        Raises:
+            RuntimeError: If loaded model has more than one image inputs
+        '''
+        super().__init__(ie, model_path, resize_type=resize_type, keep_aspect_ratio=keep_aspect_ratio)
+
+        if not self.image_blob_name:
+            raise RuntimeError("The DetectionModel wrappers supports only one image input, but {} found"
+                               .format(len(self.image_blob_names)))
+
+        if isinstance(labels, (list, tuple)):
+            self.labels = labels
+        else:
+            self.labels = load_labels(labels) if labels else None
+
+        self.threshold = threshold
+        self.iou_threshold = iou_threshold
+
+    def _resize_detections(self, detections, meta):
+        '''Resizes detection bounding box according to initial image size
+
+        Implements resize operations for different image resize types (see ``ImageModel`` class for details).
+        Applies clipping bounding box to image size.
+
+        Args:
+            detections(List[Detection]): list of detections with coordinates in normalized form
+            meta: meta information with fields `resized_shape` and `original_shape`
+
+        Returns:
+            List of detections for initial image
+
+        Raises:
+            RuntimeError: If model uses custom resize or `resize_type` not set
+        '''
+        resized_shape = meta['resized_shape']
+        original_shape = meta['original_shape']
+
+        if self.resize_type=='letterbox':
```

Comment:
That's strange you have explicit mapping for preprocess, but here you solve the same problem with if